source/extensions/common/wasm/wasm.cc

```diff
@@ -235,6 +235,53 @@ void Wasm::onStatsUpdate(const PluginSharedPtr& plugin, Envoy::Stats::MetricSnap
   context->onStatsUpdate(snapshot);
 }
 
+bool WasmHandle::restartAllowed() {
```

Comment:
rate limit decision is made on base_wasm's handle - this is the only component we share across multiple "plugins" sharing same VM.